### PR TITLE
Openshift: Don't create metricsserver apiservice

### PR DESCRIPTION
**Pull request by alvaroaleman**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Release note**:

```release-note
none
```

/assign @nikhita @zreigz

### CODE REVIEW

File: api/cmd/user-cluster-controller-manager/controllers.go
Comment (alvaroaleman) [inline api/cmd/user-cluster-controller-manager/controllers.go:37]: I feel like the idea of having a set of `addToManagerFuncs` doesn't really serve anything because different controllers do different things which implies they do or do not need different data. WDYT?

File: api/pkg/controller/usercluster/controller.go
Comment (alvaroaleman) [inline api/pkg/controller/usercluster/controller.go:23]: I have made the `Add` and the `add` one function, as the only callee of the latter is the former so I don't really see the point of having it factored out

File: api/pkg/controller/usercluster/controller.go
Comment (alvaroaleman) [inline api/pkg/controller/usercluster/controller.go:43]: TBH I don't really see the point of having a distinct `Reconciler` struct. The Controller _is_ the reconciler. Is that okay with you @nikhita ?

File: api/hack/run-userclustercontroller.sh
Comment (nikhita) [inline api/hack/run-userclustercontroller.sh:9]: In general, we should decide if we should be user-cluster or usercluster. There are still some discrepancies.

File: api/pkg/controller/usercluster/controller.go
Comment (nikhita) [reply api/pkg/controller/usercluster/controller.go:23]: Yup, good to see this. :+1: :+1: :+1:

File: api/cmd/user-cluster-controller-manager/controllers.go
Comment (zreigz) [reply api/cmd/user-cluster-controller-manager/controllers.go:37]: right now with your modifications, it doesn't have sense. The content of this file can be moved to the main

File: api/pkg/controller/usercluster/controller.go
Comment (nikhita) [reply api/pkg/controller/usercluster/controller.go:43]: Yes, I didn't really like the extra generated code by kubebuilder anyway. Happy to have it gone. :wastebasket: :heavy_check_mark:

File: api/pkg/controller/openshift/resources/usercluster.go
Comment (nikhita) [inline api/pkg/controller/openshift/resources/usercluster.go:16]: user-cluster or usercluster? :)

File: api/pkg/resources/usercluster/deployment.go
Comment (nikhita) [inline api/pkg/resources/usercluster/deployment.go:41]: Maybe I missed it -- where is this being used?

File: api/pkg/resources/usercluster/deployment.go
Comment (nikhita) [inline api/pkg/resources/usercluster/deployment.go:34]: nit: s/subet/subset

File: api/cmd/user-cluster-controller-manager/controllers.go
Comment (nikhita) [reply api/cmd/user-cluster-controller-manager/controllers.go:37]: Sgtm...we should also move the `rbacusercluster.Add` part then.

File: api/hack/run-userclustercontroller.sh
Comment (nikhita) [reply api/hack/run-userclustercontroller.sh:9]: btw @alvaroaleman this script is only used for running the controller locally, right?

Comment (alvaroaleman) [reply api/hack/run-userclustercontroller.sh:9]: Yes

File: api/pkg/controller/openshift/resources/usercluster.go
Comment (alvaroaleman) [reply api/pkg/controller/openshift/resources/usercluster.go:16]: That is what is in `pkg/resources/usercluster`

File: api/cmd/user-cluster-controller-manager/controllers.go
Comment (zreigz) [reply api/cmd/user-cluster-controller-manager/controllers.go:37]: there are two options: you can merge like it is and I will refactor RBAC part or you can refactor

**Review by zreigz — APPROVED** [review]
/lgtm

File: api/pkg/resources/usercluster/deployment.go
Comment (alvaroaleman) [reply api/pkg/resources/usercluster/deployment.go:41]: It is used transiently in the call of `apiserver.IsRunningInitContainer`

**Comment by nikhita** [comment]
/approve
/retest

/hold
I'm ok with merging and fixing the refactor stuff later. But leaving it to you. :)

**Comment by zreigz** [comment]
Friday is our refactor day so I will take care 
/hold cancel

**Pull request merged**
